cmd/erasure-server-pool-decom.go

```diff
@@ -876,7 +876,9 @@ func (z *erasureServerPools) DecommissionCancel(ctx context.Context, idx int) (e
 	defer z.poolMetaMutex.Unlock()
 
 	if z.poolMeta.DecommissionCancel(idx) {
-		z.decommissionCancelers[idx]() // cancel any active thread.
+		if len(z.decommissionCancelers) >= idx {
```

Comment:
This is not correct fix we should see why would a double cancel would panic, because it shouldn't even come here. If it does come here this shouldn't be nil. 

So the logic is a hack instead we should figure out why it is niladic in the first place.

Reply:
if decommission was tried in the past and is in canceled/failed state, upon restart pool meta is non-nil. However the context cancellation functions are never initialized since decommissioning is not in progress

Reply:
In that case this is not the correct fix @poornas - I updated the PR #14705 with correct fix, you may close this PR.